app/src/main/java/com/afzaln/awakedebug/PowerConnectionReceiver.java

```diff
@@ -32,8 +32,16 @@ public static void toggleStayAwake(Context context) {
         Intent batteryStatus = context.getApplicationContext().registerReceiver(null, filter);
 
         int chargePlug = batteryStatus.getIntExtra(BatteryManager.EXTRA_PLUGGED, -1);
-        boolean usbCharge = chargePlug == BatteryManager.BATTERY_PLUGGED_USB ||
-                (chargePlug == BatteryManager.BATTERY_PLUGGED_AC && Utils.getAcPowerOn(context));
+        boolean usbCharge;
+
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.JELLY_BEAN_MR1) {
+            usbCharge = chargePlug == BatteryManager.BATTERY_PLUGGED_USB ||
+                    chargePlug == BatteryManager.BATTERY_PLUGGED_WIRELESS ||
```

Comment:
Most definitely do not want the device the keep awake during wireless charging without a third explicit option for it because wireless chargers aren't fast enough to sustain a charge in that state.